File: 021 - Amicable numbers/21.c
#include <stdio.h>

unsigned int sum_of_divisors1, sum_of_divisors2, num, i, amicable_sum=0;

main(){

    for(num=1; num<10000; num++){

        sum_of_divisors1=0;
        sum_of_divisors2=0;

        i=1;
        while(i<num){
            if(num%i==0)
                sum_of_divisors1+=i; //sum of the devisors of num
            ++i;
        }

        i=1;
        while(i<sum_of_divisors1){
            if(sum_of_divisors1%i==0)
                sum_of_divisors2+=i; //sum of the devisors of sum_of_divisors1
            ++i;
        }

        if(sum_of_divisors2==num && sum_of_divisors1!=sum_of_divisors2){ // check whether amicable
            printf("Amicable pairs %d, %d\n\n", sum_of_divisors1, sum_of_divisors2);
            amicable_sum = amicable_sum + sum_of_divisors1 + sum_of_divisors2; // calculate sum of all amicable pairs
        }

    }

    printf("Sum of all the amicable numbers under 10000 = %d\n", amicable_sum/2);

}

File: project_euler/021 - Amicable numbers/21.py

amicable = set()

for i in range(1,10000):

    if i not in amicable:

        sum1 = 0
        sum2 = 0
        
        print i

        a = 1
        while a<i:
            if i%a==0:
                sum1+=a
            a+=1

        a = 1
        while a<sum1:
            if sum1%a==0:
                sum2+=a
            a+=1

        if sum2==i and sum1!=sum2:
            print "Amicable pairs :", sum1, sum2
            amicable.add(sum1)
            amicable.add(sum2)

print "Sum of all amicable numbers under 100 =", sum(amicable)
    
